www/docs/server/context.md

```diff
@@ -60,11 +62,88 @@ export const router = t.router;
 
 /**
  * Unprotected procedure
- **/
+ */
 export const publicProcedure = t.procedure;
 
 /**
  * Protected procedure
- **/
+ */
 export const protectedProcedure = t.procedure.use(isAuthed);
-```
\ No newline at end of file
+```
+
+## Inner and outer context
+
+In some scenarios it could make sense to split up your context into "inner" and "outer" functions.
+
+**Inner context** is where you define context which doesn’t depend on the request, e.g. your database connection. You can use this function for integration testing or [SSG helpers](ssg-helpers), where you don’t have a request object. Whatever is defined here will **always** be available in your procedures.
+
+**Outer context** is where you define context which depends on the request, e.g. for the user's session. Whatever is defined here is not available in all procedures, but only for procedures that are called via HTTP.
```

Comment:
Instead of "not available in all procedures", which makes it sound like it somehow depends on the type of procedure, I'd suggest to continue the explanation from the previous paragraph and say something like "only available when a procedure is called via http".